Readonly tuples aren't considered arrays

[Playground](https://typescripttolua.github.io/play.html#src=const%20foo%20%3D%20%5B1%2C%202%5D%20as%20const%3B%0Afoo%5B0%5D%3B%0A)

It seems like we're not using a new enough version of TS to support this?
```ts
const foo = [1, 2] as const; // error TS2304: Cannot find name 'const'.
foo[0];
```

It looks like we're using 3.5.2 which does support `as const`, but TSTL still throws this error when I try to compile.

--- a/src/TSHelper.ts
+++ b/src/TSHelper.ts
@@ -167,12 +167,12 @@ export function isExplicitArrayType(type: ts.Type, checker: ts.TypeChecker, prog
         return type.types.some(t => isExplicitArrayType(t, checker, program));
     }
 
-    if (isStandardLibraryType(type, "ReadonlyArray", program)) {
-        return true;
+    const flags = ts.NodeBuilderFlags.InTypeAlias | ts.NodeBuilderFlags.AllowEmptyTuple;
+    let typeNode = checker.typeToTypeNode(type, undefined, flags);
+    if (typeNode && ts.isTypeOperatorNode(typeNode) && typeNode.operator === ts.SyntaxKind.ReadonlyKeyword) {
+        typeNode = typeNode.type;
     }
 
-    const flags = ts.NodeBuilderFlags.InTypeAlias | ts.NodeBuilderFlags.AllowEmptyTuple;
-    const typeNode = checker.typeToTypeNode(type, undefined, flags);
     return typeNode !== undefined && (ts.isArrayTypeNode(typeNode) || ts.isTupleTypeNode(typeNode));
 }
 

--- a/test/unit/tuples.spec.ts
+++ b/test/unit/tuples.spec.ts
@@ -31,6 +31,15 @@ test("Tuple access", () => {
     expect(result).toBe(5);
 });
 
+test("Readonly Tuple access", () => {
+    const result = util.transpileAndExecute(
+        `const tuple: readonly [number, number, number] = [3,5,1];
+        return tuple[1];`
+    );
+
+    expect(result).toBe(5);
+});
+
 test("Tuple union access", () => {
     const result = util.transpileAndExecute(
         `function makeTuple(): [number, number, number] | [string, string, string] { return [3,5,1]; }

